controllers/provider-azure/pkg/apis/azure/types_infrastructure.go

```diff
@@ -23,6 +23,7 @@ import (
 // InfrastructureConfig infrastructure configuration resource
 type InfrastructureConfig struct {
 	metav1.TypeMeta
+	metav1.ObjectMeta
```

Comment:
Why do we now get object meta here?

Reply:
Hm, I don't understand. The `Infrastructure` resource is defined in Gardener's extension API, and here you define an embedded `InfrastructureConfig` resource that is part of the `Infrastructure` object. Why does it have to have the namespace again? That's not convenient for the end-user and not needed. Can't you simply pass the namespace name to your validation function?

Reply:
Yep, I agree. I have modified it to pass the shoot resourceGroup name (=shoot technical name) to the Validation function. Thank you!